packages/eslint-plugin/src/rules/prefer-reduce-type-parameter.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as ts from 'typescript';
+import * as util from '../util';
+
+const getMemberExpressionName = (
+  member: TSESTree.MemberExpression | TSESTree.OptionalMemberExpression,
+): string | null => {
+  if (!member.computed) {
+    return member.property.name;
+  }
+
+  if (
+    member.property.type === AST_NODE_TYPES.Literal &&
+    typeof member.property.value === 'string'
+  ) {
+    return member.property.value;
+  }
+
+  return null;
+};
+
+export default util.createRule({
+  name: 'prefer-reduce-type-parameter',
+  meta: {
+    type: 'problem',
+    docs: {
+      category: 'Best Practices',
+      recommended: 'error',
```

Comment:
nit - can't make it recommended without a major release
```suggestion
      recommended: false,
```